File: cwclock-ui/src/Components/common/memberLabel.js
// Renders a member's display name, falling back to their email when no
// name/surname has been set yet (eg. members who signed up before this field existed).
const memberLabel = (member) => {
  return member.name || member.surname ? `${member.name || ""} ${member.surname || ""}`.trim() : member.email;
}

export default memberLabel;

File: cwclock-mobile/lib/common/member_label.dart
import '../models/member.dart';

/// Renders a member's display name, falling back to their email when no
/// name/surname has been set yet. Mirrors cwclock-ui's memberLabel.js.
String memberLabel(Member member) {
  return member.name?.isNotEmpty == true || member.surname?.isNotEmpty == true
      ? '${member.name ?? ''} ${member.surname ?? ''}'.trim()
      : member.email;
}
